packages/relay/src/lib/eth.ts

```diff
@@ -1455,6 +1470,49 @@ export class EthImpl implements Eth {
     }
   }
 
+  /**
+   * Sends a raw transaction with retry logic.
+   *
+   * @template T The type of the result returned by the sendRawTransaction function.
+   * @param {Function} sendRawTransaction The function responsible for sending the raw transaction.
+   * @param {object} options The options for retrying the transaction.
+   * @param {number} [options.maxAttempts=2] The maximum number of attempts to send the transaction.
+   * @param {number} [options.backOff=500] The backoff period in milliseconds between retry attempts.
+   * @param {(error: unknown) => boolean} [options.canRetry=(error) => true] A function that determines whether a retry attempt can be made based on the error received.
+   * @param {(error: SDKClientError) => void} [options.onError=(error) => {}] A function to handle errors that occur during retry attempts.
+   * @returns {Promise<T | undefined>} A promise resolving to the result of the transaction, or undefined if all retry attempts fail.
+   */
+  private async sendRawTransactionWithRetry<T>(
+    sendRawTransaction: () => T,
+    {
+      maxAttempts = 2,
+      backOff = 500,
+      canRetry = (e: unknown) => true,
+      onError = (e: SDKClientError) => {},
+    }: {
+      maxAttempts?: number;
+      backOff?: number;
+      canRetry?: (error: unknown) => boolean;
+      onError?: (error: SDKClientError) => void;
+    },
+  ): Promise<T | undefined> {
+    for (let count = 0; count < maxAttempts; count++) {
+      try {
+        return await sendRawTransaction();
+      } catch (e: unknown) {
+        if (!canRetry(e) || count === maxAttempts - 1) {
+          throw e;
+        }
+        onError(e as SDKClientError);
+        // eslint-disable-next-line no-await-in-loop
+        await new Promise((resolve) => {
```

Comment:
A small suggestive change - move this to a function called delay. I think it would be more descriptive and clear about what is happening here
Something like
`const delay = backOff => new Promise(resolve => setTimeout(resolve, backOff));`

Reply:
Took out in a variable 👍